cmd/object-handlers.go

```diff
@@ -788,13 +788,14 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		length = actualSize
 	}
 
+	var compressMetadata = make(map[string]string)
```

Comment:
you only need to allocate this if `isCompressed` is true so start with 

```
var compressMetadata map[string]string

if isCompressed {
    compressMetadata = make(map[string]string)
}

for k, v := range compressMetadata {
}
```

Reply:
Done :+1: